tests/alfacase/test_case_to_alfacase.py

```diff
@@ -44,6 +49,39 @@ def test_convert_description_to_alfacase_with_empty_dict(tmp_path):
     assert loaded_alfacase.content["name"].data == simple_case.name
 
 
+@pytest.mark.parametrize(
+    "remove", [pytest.param(True, id="remove"), pytest.param(False, id="keep")]
+)
+@pytest.mark.parametrize("input_type", [MultiInputType.Constant, MultiInputType.Curve])
+def test_remove_redundant_input_type_data_option(
+    remove: bool, input_type: MultiInputType
+) -> None:
+    mass_source_equipment_description = case_description.MassSourceEquipmentDescription(
+        position=Scalar(0, "m"), temperature_input_type=input_type
+    )
+    yaml = convert_description_to_alfacase(
+        mass_source_equipment_description, remove_redundant_input_type_data=remove
+    )
+
+    def has_key_in_yaml(key: str) -> bool:
+        pattern = rf"^\s*{re.escape(key)}\s*:"
+        return bool(re.search(pattern, yaml, re.MULTILINE))
+
+    if remove:
+        assert not has_key_in_yaml("temperature_input_type")
+        if input_type == MultiInputType.Constant:
+            assert has_key_in_yaml("temperature")
+            assert not has_key_in_yaml("temperature_curve")
+        elif input_type == MultiInputType.Curve:
+            assert not has_key_in_yaml("temperature")
+            assert has_key_in_yaml("temperature_curve")
+
+    else:
+        assert has_key_in_yaml("temperature_input_type")
+        assert has_key_in_yaml("temperature")
+        assert has_key_in_yaml("temperature_curve")
```

Comment:
Hmmm we have an empty key in there, or a key with the defaul value?

Reply:
Actually this section of the test is just to cover this "feature" in a unit test.
It dumps all multi input fields to the `.alfacase` file, the usual is just to dump the currently used.
The full dump of multi input field was implemented to make round trip tests easier.

TL;DR;
This is the info the is present in the case description, because they are optional they also are required to have a default value. If the exported case description have not set those values the default will be present.